evm/src/main/java/org/hyperledger/besu/evm/fluent/SimpleWorld.java

```diff
@@ -70,45 +72,46 @@ public MutableAccount createAccount(final Address address, final long nonce, fin
       throw new IllegalStateException("Cannot create an account when one already exists");
     }
     SimpleAccount account = new SimpleAccount(address, nonce, balance);
-    accounts.put(address, account);
+    accounts.put(address, Optional.of(account));
     return account;
   }
 
   @Override
   public MutableAccount getAccount(final Address address) {
-    SimpleAccount account = accounts.get(address);
+    Optional<SimpleAccount> account = accounts.get(address);
     if (account != null) {
-      return account;
+      return account.orElse(null);
     }
     Account parentAccount = parent == null ? null : parent.getAccount(address);
     if (parentAccount != null) {
       account =
-          new SimpleAccount(
-              parentAccount,
-              parentAccount.getAddress(),
-              parentAccount.getNonce(),
-              parentAccount.getBalance(),
-              parentAccount.getCode());
+          Optional.of(
+              new SimpleAccount(
+                  parentAccount,
+                  parentAccount.getAddress(),
+                  parentAccount.getNonce(),
+                  parentAccount.getBalance(),
+                  parentAccount.getCode()));
       accounts.put(address, account);
-      return account;
+      return account.get();
     }
     return null;
   }
 
   @Override
   public void deleteAccount(final Address address) {
-    accounts.put(address, null);
+    accounts.put(address, Optional.empty());
   }
 
   @Override
   public Collection<? extends Account> getTouchedAccounts() {
-    return accounts.values();
+    return accounts.values().stream().filter(Optional::isPresent).map(Optional::get).toList();
```

Comment:
similar comment regarding deleted accounts.  a deleted account is a touched account.  

I am not sure if we want to indicate an eip-6780 transient account delete is a touched account though 🤔

Reply:
Not sure what you mean in this particular method - is it about not returning the same collection size anymore? Apart from that I don't get why returning nulls in the collection is useful

Reply:
6780 transient deletes should never hit the simple world. The stacked or journaled updates above the simple world will elide it out.  Only OG SELFDESTRUCTSs will be seen through to the SimpleWorld.  Given the fluent API supports all forks, we need to be able to support that.

Reply:
After looking at the usages of  getTouchedAccounts(), it seems we only use this interface in the context of accounts which will continue to exist after the commit, those which would be deleted (including eip-6780 self destructs) are always accessed via getDeletedAccounts.  The interface comment is misleading (or overloads the meaning for 'touched')

In the context of bonsai, trie logs, linea, etc - all touched accounts includes read, updated, and deleted.  But that isn't the context here.  

In short, LGTM :)